module-telemetry/src/main/java/com/mapbox/maps/module/telemetry/MapTelemetryImpl.kt

```diff
@@ -28,39 +34,78 @@ class MapTelemetryImpl : MapTelemetry {
   constructor(appContext: Context, accessToken: String) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = MapboxTelemetry(appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT)
-    val telemetryState: TelemetryEnabler.State =
-      TelemetryEnabler.retrieveTelemetryStateFromPreferences()
-    if (TelemetryEnabler.State.ENABLED == telemetryState) {
-      telemetry.enable()
+
+    // check in the resources
+    val eventsTokenResId = getEventsResId(appContext, EVENTS_ACCESS_TOKEN_RESOURCE_NAME)
+    val eventsToken =
+      if (eventsTokenResId != 0) appContext.getString(eventsTokenResId) else accessToken
+
+    // TODO deferredDeliveryServiceOptions?
+    val eventsServiceOptions = EventsServerOptions(eventsToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
+    this.eventsService = EventsService.getOrCreate(eventsServiceOptions)
+
+    this.telemetryService = TelemetryService.getOrCreate(eventsServiceOptions)
+
+    val coreTelemetryState = TelemetryUtils.getEventsCollectionState()
+    if (coreTelemetryState) {
+      enableTelemetryCollection(true)
```

Comment:
we had more tricky code with pause/resume collection and simultaneous management of MME & Core Telemetry here, but now we do not need 

```
if (coreTelemetryState) {
      enableTelemetryCollection(true)
```
cause it is 
```
if (TelemetryUtils.getEventsCollectionState() == true)
  TelemetryUtils.setEventsCollectionState(true)
```

Reply:
enableTelemetryCollection is also used in several other places, but I've removed `coreTelemetryState` variable.